商品詳細表示機能実装

gucchi-ta opened the pull request:
# WHAT
商品詳細表示機能の実装

# WHY
商品詳細表示機能実装のため

# GyazoURL
- ログインしたユーザーと出品者が一致している場合は編集・削除ボタンを表示
(一覧ページの商品画像クリックで詳細ページへ遷移)
https://gyazo.com/46912b028f8127b99f4d905ab29c51c9
- 商品が未売却の場合は購入ボタンを表示(1枚目出品者と別のユーザーでログインした時 /2 枚目未ログインの時)
https://gyazo.com/2847ed2d32aede47267a73f4b7efb502
https://gyazo.com/523bdd472a74b10a53863bb4ce11322a
- 商品が売却済みの場合は購入ボタンが表示されない(1枚目出品者と別のユーザーでログインした時 / 2枚目未ログインの時)
https://gyazo.com/5bd5c3b59b6df31af10a5b65bc4e6826
https://gyazo.com/e06a9b28e40fe0902b9d833974c1b03b
- 商品の情報を詳細表示画面に表示
https://gyazo.com/cf50384ea1a16fa90b3ffda7bb1206cf
- rubokopの実行
https://gyazo.com/5e1584a45fd129e96134a7dc060cc840

gucchi-ta committed:
ビューファイルの配置/詳細画面への遷移

gucchi-ta committed:
商品詳細表示機能の実装

gucchi-ta committed:
(edit.html.erbの配置)

kokikurahashi commented on app/controllers/items_controller.rb:
```diff
@@ -24,10 +19,21 @@ def create
    end
 
+   def show
+      @item = Item.find(params[:id])
```
こちらの記述は他のアクションでも使用する可能性の高いコードになりますのでメソッド化し、before_actionで定義するようにしましょう！！

gucchi-ta replied:
修正しました!
(以下rubocopした際のGyazoURLです)
https://gyazo.com/8e13f83d57e4ae50c02ad273b89792a3

gucchi-ta committed:
rubokopの実行と修正

rubocopの実行の際
bundle exec rubocop -lをしていたが
bundle exec rubocop -aを使い修正後
bundle exec rubocopを実行

gucchi-ta committed:
編集/削除ボタン表示の条件を変更

kokikurahashi reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

gucchi-ta committed:
複数回使う記述をメソッドとして定義

shingoteshima commented:
レビュー依頼ありがとうございます！
LGTMです！！
お疲れさまでした！

gucchi-ta commented:
ありがとうございます!
引き続き実装進めて参ります！
